src/Libraries/RevitNodes/Elements/Parameter.cs

```diff
@@ -40,5 +165,244 @@ public override string ToString()
             }
             return string.Format("{0} : {1}", Name, value);
         }
+
+        #endregion
+
+        #region Shared Parameters
+
+        /// <summary>
+        /// Gets the path to the shared parameter file of this document
+        /// </summary>
+        /// <returns></returns>
+        public static string SharedParameterFile()
+        {
+            Document document = Application.Document.Current.InternalDocument;
+            return document.Application.SharedParametersFilename;
+        }
+
+        /// <summary>
+        /// Create a new Shared Parameter in the current Revit document for all applicable categories
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="type">Parameter Type</param>
+        /// <param name="group">Parameter Group</param>
+        /// <param name="instance">Is instance parameter, otherwise its a type parameter</param>
+        public static void CreateSharedParameter(string parameterName, string groupName, string type, string group, bool instance)
+        {
+            CreateSharedParameter(parameterName, groupName, type, group, instance, null); 
+        }
+
+        /// <summary>
+        /// Create a new Shared Parameter in the current Revit document
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="type">Parameter Type</param>
+        /// <param name="group">Parameter Group</param>
+        /// <param name="instance">Is instance parameter, otherwise its a type parameter</param>
+        /// <param name="categoryList">List of categories this parameter applies to, If no category is supplied, all possible categories are selected</param>
+        public static void CreateSharedParameter(string parameterName, string groupName, string type, string group, bool instance, System.Collections.Generic.IEnumerable<Category> categoryList)
```

Comment:
I see about the overload now I did not realize it was for the public method, is there any issue with actually getting rid of the public overloads? Can you simply change the name of this method or the other one?